compiler/rustc_symbol_mangling/src/test.rs

```diff
@@ -59,16 +60,27 @@ impl SymbolNamesTest<'_> {
                 tcx.erase_regions(InternalSubsts::identity_for_item(tcx, def_id)),
             );
             let mangled = tcx.symbol_name(instance);
-            tcx.sess.span_err(attr.span, &format!("symbol-name({})", mangled));
+            tcx.sess.emit_err(InvalidSymbolName {
+                span: attr.span,
+                mangled_formatted: format!("{mangled}"),
+            });
             if let Ok(demangling) = rustc_demangle::try_demangle(mangled.name) {
-                tcx.sess.span_err(attr.span, &format!("demangling({})", demangling));
-                tcx.sess.span_err(attr.span, &format!("demangling-alt({:#})", demangling));
+                tcx.sess.emit_err(InvalidTraitItem {
+                    span: attr.span,
+                    demangling_formatted: format!("{demangling}"),
+                });
+                tcx.sess.emit_err(AltInvalidTraitItem {
+                    span: attr.span,
+                    alt_demangling_formatted: format!("{:#}", demangling),
+                });
             }
         }
 
         for attr in tcx.get_attrs(def_id.to_def_id(), DEF_PATH) {
-            let path = with_no_trimmed_paths!(tcx.def_path_str(def_id.to_def_id()));
-            tcx.sess.span_err(attr.span, &format!("def-path({})", path));
+            tcx.sess.emit_err(InvalidDefPath {
+                span: attr.span,
+                def_path: with_no_trimmed_paths!(tcx.def_path_str(def_id.to_def_id())),
+            });
```

Comment:
I was confused why we had symbol-mangling diagnostics... and we don't.

This probably needs to be reverted, or at the very least fixed to not have very misleading names.

There are no user-facing errors here, these are the `#[rustc_symbol_name]` and `#[rustc_def_path]` *internal testing* attributes (similar to `#[rustc_layout]` which should also not be considered user-facing).

Even these using `emit_err` is a bit of a stretch, but we do need to point at the source code so we can't just dump something on stderr (there are more reasons for that as well).

Reply:
At the very least, I would like to see the name of the the attribute generating this (instead of "invalid").

So e.g. this should be `rustc_def_path` instead of `invalid_def_path`.

Ideally it would be tagged as "internal" so translators don't waste their time or get confused - there's basically no natural-language text here, the `def-path(...)` can just be changed to mentioning `#[rustc_def_path]` instead, since it only exists to label the test output - or could even just remove the text around the `tcx.def_path_str(...)` result entirely.

I would compare it to `RUSTC_LOG` and `#[tracing::instrument]`: you shouldn't translate it, because it's just reflecting things in the compiler *and* it's not user-facing.
<sub>(`debug!` is a bit different and it would be interesting to see what we can do in terms of trying to remove natural-language messaging from them, and maybe long-term translate the bits that we can't - but that's getting close to "translating the comments in the compiler", or even identifiers, and at that point a better solution might be a way to annotate the source-code as a whole externally, or at least it's a harder problem I don't know of good solutions for)</sub>